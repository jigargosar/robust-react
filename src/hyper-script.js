import h from 'react-hyperscript'
import hh from 'hyperscript-helpers'

export const {div} = hh(h)


export {h}
