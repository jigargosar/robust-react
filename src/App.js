import {
  AppBar,
  CssBaseline,
  List,
  MenuItem,
  Paper,
  TextField,
  Toolbar,
  Typography,
  withStyles,
} from 'material-ui'
import {div, h} from './hyper-script'
import S from './sanctuary'

const NumberList = () =>
  h(
    List,
    S.pipe(
      [S.range(0), S.reverse, S.map(id => h(MenuItem, {key: id}, id + 1))],
      20,
    ),
  )

const NumberListLayout = withStyles(theme => ({
  root: {
    overflow: 'scroll',
    flex: 1,
  },
  container: {
    display: 'flex',
    flexDirection: 'column-reverse',
  },
  paper: {margin: theme.spacing.unit * 2},
}))(({classes}) =>
  div({className: classes.root}, [
    div({className: classes.container}, [
      h(Paper, {className: classes.paper}, [h(NumberList)]),
    ]),
  ]),
)

const Header = withStyles(({spacing: {unit}}) => ({
  flex: {
    flex: 1,
  },
  textFieldRoot: {
    // backgroundColor: 'rgba(0,0,0,0.1)',
    backgroundColor: 'rgba(255,255,255,0.1)',
    color: 'inherit',
  },
  textFieldInput: {
    margin: `0 ${unit}px`,
  },
}))(({classes}) =>
  h(AppBar, {position: 'static'}, [
    h(Toolbar, [
      h(
        Typography,
        {
          className: classes.flex,
          variant: 'title',
          color: 'inherit',
        },
        'Robust React App',
      ),
      h(TextField, {
        type: 'search',
        placeholder: 'Search',
        InputProps: {
          disableUnderline: true,
          classes: {
            root: classes.textFieldRoot,
            input: classes.textFieldInput,
          },
        },
      }),
    ]),
  ]),
)
const Footer = withStyles({})(() =>
  h(AppBar, {position: 'static', component: 'footer'}, [
    h(Toolbar, [h(Typography, {variant: 'title', color: 'inherit'}, 'Footer')]),
  ]),
)
const Layout = withStyles({
  root: {
    height: '100vh',
    width: '100vw',
    overflow: 'hidden',
    display: 'flex',
    flexDirection: 'column',
  },
})(({children, classes}) => div({className: classes.root}, [children]))

const App = () =>
  h(CssBaseline, [h(Layout, [h(Header), h(NumberListLayout), h(Footer)])])

export default App
