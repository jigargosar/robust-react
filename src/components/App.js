import {
  AppBar,
  CssBaseline,
  Paper,
  Toolbar,
  Typography,
  withStyles,
} from 'material-ui'
import {div, h} from '../hyper-script'
import Header from './Header'
import ModelList from './model/List'

const ModelListLayout = withStyles(theme => ({
  root: {
    overflow: 'scroll',
    flex: 1,
  },
  paper: {
    margin: theme.spacing.unit * 2,
  },
}))(({classes}) =>
  div({className: classes.root}, [
    h(Paper, {className: classes.paper}, [h(ModelList)]),
  ]),
)

const Footer = withStyles({})(() =>
  h(AppBar, {position: 'static', component: 'footer'}, [
    h(Toolbar, [h(Typography, {variant: 'title', color: 'inherit'}, 'Footer')]),
  ]),
)
const Layout = withStyles({
  root: {
    height: '100vh',
    width: '100vw',
    overflow: 'hidden',
    display: 'flex',
    flexDirection: 'column',
  },
})(({children, classes}) => div({className: classes.root}, [children]))

const App = () =>
  h(CssBaseline, [h(Layout, [h(Header), h(ModelListLayout), h(Footer)])])

export default App
