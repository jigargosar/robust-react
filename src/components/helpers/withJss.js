import {createGenerateClassName} from 'material-ui/styles'
import JssProvider from 'react-jss/lib/JssProvider'
/* eslint-disable babel/new-cap */
import {render} from 'react-testing-library'
import {h} from '../../hyper-script'
import S from '../../sanctuary'

const generateClassName1 = (rule, styleSheet) =>
  `${styleSheet.options.classNamePrefix}-${rule.key}`

const generateClassName2 = createGenerateClassName({
  dangerouslyUseGlobalCSS: true,
})

const generateClassName = S.I(true) ? generateClassName1 : generateClassName2
const withJss = component =>
  render(
    h(JssProvider, {generateClassName}, component),
    // component,
  )
export default withJss
