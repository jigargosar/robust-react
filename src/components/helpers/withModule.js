import {Container} from '@cerebral/react'
import {Controller} from 'cerebral'
import {Module} from 'cerebral/index'
import {h} from '../../hyper-script'

const withModule = (module, component) =>
  h(Container, {controller: Controller(Module(module))}, [component])

export default withModule
