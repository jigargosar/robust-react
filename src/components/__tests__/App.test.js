import {h} from '../../hyper-script'
import App from '../App'
import withJss from '../helpers/withJss'
import withModule from '../helpers/withModule'

it('renders without crashing', () => {
  const {container} = withJss(withModule({}, h(App)))
  expect(container.firstChild).toMatchSnapshot()
})
