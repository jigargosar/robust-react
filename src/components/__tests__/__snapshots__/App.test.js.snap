// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  class="Component-root"
>
  <header
    class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters"
    >
      <h2
        class="MuiTypography-root MuiTypography-title MuiTypography-colorInherit Header-flex"
      >
        Robust React App
      </h2>
      <div
        class="MuiFormControl-root"
      >
        <div
          class="MuiInput-root Search-root MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            aria-required="false"
            class="MuiInput-input Search-input MuiInput-inputType MuiInput-inputTypeSearch"
            placeholder="Search"
            type="search"
            value=""
          />
        </div>
      </div>
    </div>
  </header>
  <div
    class="Component-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation2 MuiPaper-rounded Component-paper"
    >
      <ul
        class="MuiList-root MuiList-padding"
      >
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          Philippines
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          Puerto Rico
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          Eritrea
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          Ireland
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          Côte d'Ivoire
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          Paraguay
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          Marshall Islands
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          Qatar
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          Réunion
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          Lebanon
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          Iraq
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          Slovenia
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          Wallis & Futuna
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          Iceland
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          Papua New Guinea
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          Moldova
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          Kyrgyzstan
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          Burkina Faso
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          Haiti
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          Hong Kong SAR China
          <span
            class="MuiTouchRipple-root"
          />
        </li>
      </ul>
    </div>
  </div>
  <footer
    class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters"
    >
      <h2
        class="MuiTypography-root MuiTypography-title MuiTypography-colorInherit"
      >
        Footer
      </h2>
    </div>
  </footer>
</div>
`;
